frontend/src/lib/utils.tsx

```diff
@@ -511,16 +511,16 @@ export const dateMapping: Record<string, string[]> = {
 export const isDate = /([0-9]{4}-[0-9]{2}-[0-9]{2})/
 
 export function dateFilterToText(
-    dateFrom: string | dayjs.Dayjs | undefined,
-    dateTo: string | dayjs.Dayjs | undefined,
+    dateFrom: string | dayjs.Dayjs | null | undefined,
+    dateTo: string | dayjs.Dayjs | null | undefined,
     defaultValue: string
 ): string {
     if (dayjs.isDayjs(dateFrom) && dayjs.isDayjs(dateTo)) {
         return `${dateFrom.format('YYYY-MM-DD')} - ${dateTo.format('YYYY-MM-DD')}`
     }
-    dateFrom = dateFrom as string
-    dateTo = dateTo as string
-    if (isDate.test(dateFrom) && isDate.test(dateTo)) {
+    dateFrom = (dateFrom || undefined) as string | undefined
+    dateTo = (dateTo || undefined) as string | undefined
+    if (isDate.test(dateFrom || '') && isDate.test(dateTo || '')) {
```

Comment:
How come we don't set the dateFrom/dateTo variables to `dateFrom || ''` and  `dateTo || ''` on line 521 and 522 respectively instead of when we pass it in here?

Reply:
It might mess with iteration on lines 529-535.

However, don't think this matters that much. :)